arkane/reference.py

```diff
@@ -0,0 +1,430 @@
+            raise KeyError('Model chemistry `{0}` not available for species {1}'.format(model_chemistry, self))
+        if not self.reference_data:
+            raise ValueError('No reference data is included for species {0}'.format(self))
+
+        molecule = Molecule(SMILES=self.smiles)
+        preferred_source = source
+
+        if not preferred_source:
+            # Find the preferred source
+            if self.preferred_reference is not None:
+                preferred_source = self.preferred_reference
+            else:  # Choose the source that has the smallest uncertainty
+                sources = self.reference_data.keys()
+                data = self.reference_data.values()
+                preferred_source = sources[0]  # If all else fails, use the first source as the preferred one
+                uncertainty = data[0].thermo_data.H298.uncertainty_si
+                for i, entry in enumerate(data):
+                    if (entry.thermo_data.H298.uncertainty_si > 0) and \
+                            (entry.thermo_data.H298.uncertainty_si < uncertainty):
+                        uncertainty = entry.thermo_data.H298.uncertainty_si
+                        preferred_source = sources[i]
+        high_level_h298 = self.reference_data[preferred_source].thermo_data.H298.__reduce__()[1]
+        low_level_h298 = self.calculated_data[model_chemistry].thermo_data.H298.__reduce__()[1]
+
+        return ErrorCancelingSpecies(molecule, low_level_h298, model_chemistry, high_level_h298, preferred_source)
+
+
+class ReferenceDataEntry(RMGObject):
+    """
+    A class for storing reference data for a specific species from a single source
+    """
+    def __init__(self, thermo_data, atct_id=None):
+        """
+
+        Args:
+            thermo_data (rmgpy.thermo.ThermoData): Thermochemistry (Hf298, Cp, ...) from the reference for a species
+            atct_id (str): ID number in the Active Thermochemical Tables if the source is ATcT
+        """
+        super(ReferenceDataEntry, self).__init__()
+        self._thermo_data = None
+        self.thermo_data = thermo_data
+        self.atct_id = atct_id
+
+    def __repr__(self):
+        return str(self.as_dict())
+
+    @property
+    def thermo_data(self):
+        return self._thermo_data
+
+    @thermo_data.setter
+    def thermo_data(self, value):
+        if value:
+            if isinstance(value, ThermoData):
+                self._thermo_data = value
+            else:
+                raise ValueError('thermo_data for a ReferenceDataEntry object must be an rmgpy ThermoData instance')
+        else:
+            self._thermo_data = None
+
+
+class CalculatedDataEntry(RMGObject):
+    """
+    A class for storing a single entry of statistical mechanical and thermochemistry information calculated at a single
+    model chemistry or level of theory
+    """
+    def __init__(self, conformer, thermo, thermo_data, t1_diagnotic=None, fod=None):
+        """
+
```

Comment:
typo in argument `t1_diagnotic` -> `t1_diagnostic`

Reply:
Fixed, thanks!